pkgs/build-support/fetchpijul/default.nix

```diff
@@ -0,0 +1,57 @@
+{ stdenvNoCC, pijul, lib }:
+
+lib.makeOverridable (
+{ url
+, sha256 ? ""
+, hash ? ""
+, change ? null
+, state ? null
+, channel ? "main"
+, name ? "fetchpijul"
+, # TODO: Changes in pijul are unordered so there's many ways to end up with the same repository state.
+  # This makes leaveDotPijul unfeasible to implement until pijul CLI implements
+  # a way of reordering changes to sort them in a consistent and deterministic manner.
+  # leaveDotPijul ? false
+}:
+if change != null && state != null then
+  throw "Only one of change or state can be set"
+else if hash != "" && sha256 != "" then
+  throw "Only one of sha256 or hash can be set"
+else
+  stdenvNoCC.mkDerivation {
+    inherit name;
+    nativeBuildInputs = [ pijul ];
+
+    dontUnpack = true;
+
+    installPhase = ''
+      runHook preInstall
+
+      pijul clone \
```

Comment:
(re: message in matrix) I think this is fine in installPhase, at least the "putting stuff in $out" part fits here, and splitting it up to only move files to $out here seems like unnecessarily making it more complicated